### PR TITLE
Document the user endpoints

### PR DESCRIPTION
**Pull request by rv0lt**
## Read this before submitting the PR

1. Always create a Draft PR first
2. Go through sections 1-5 below, fill them in and check all the boxes
3. Make sure that the branch is updated; if there's an "Update branch" button at the bottom of the PR, rebase or update branch.
4. When all boxes are checked, information is filled in, and the branch is updated: mark as Ready For Review and tag reviewers (top right)
5. Once there is a submitted review, implement the suggestions (if reasonable, otherwise discuss) and request an new review.

If there is a field which you are unsure about, enter the edit mode of this description or go to the [PR template](../.github/pull_request_template.md); There are invisible comments providing descriptions which may be of help.

## 1. Description / Summary

Added in the swagger file the documentation for the user endpoints

## 2. Jira task / GitHub issue

https://scilifelab.atlassian.net/jira/software/projects/DDS/boards/13?selectedIssue=DDS-1849

## 3. Type of change

What _type of change(s)_ does the PR contain?

**Check the relevant boxes below. For an explanation of the different sections, enter edit mode of this PR description template.**

- [ ] New feature
  - [ ] Breaking: _Why / How? Add info here._ 
  - [ ] Non-breaking 
- [ ] Database change: _Remember the to include a new migration version, **or** explain here why it's not needed._ 
- [ ] Bug fix 
- [ ] Security Alert fix 
  - [ ] Package update 
    - [ ] Major version update 
- [x] Documentation 
- [ ] Workflow 
- [ ] Tests **only** 

## 4. Additional information

- [x] [Sprintlog](../SPRINTLOG.md) 
- [ ] Blocking PRs 
  - [ ] Merged 
- [ ] PR to `master` branch: \_If checked, read [the release instructions](../doc/procedures/new_release.md) 
  - [ ] I have followed steps 1-8. 

## 5. Actions / Scans

_Check the boxes when the specified checks have passed._

**For information on what the different checks do and how to fix it if they're failing, enter edit mode of this description or go to the [PR template](../.github/pull_request_template.md).**

- [x] **Black**

- [x] **Prettier**

- [x] **Yamllint**

- [x] **Tests**

- [x] **CodeQL**

- [x] **Trivy**

- [x] **Snyk**

### CODE REVIEW

File: dds_web/static/swagger.yaml [hunk 1]
```diff
@@ -466,7 +466,339 @@ paths:
...
+        "500":
+          $ref: "#/components/responses/InternalServerlError"
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: string
+                example: "{message: Requested account deletion initiated.
+                  An e-mail with a confirmation link has been sent to your address}"
```
Comment (valyo) [inline dds_web/static/swagger.yaml:606]:
```suggestion
                example: "{message: User with email email@example.com no longer associated with exmple_project.}"
```

File: dds_web/static/swagger.yaml [hunk 2]
```diff
@@ -631,6 +980,85 @@ components:
...
+          type: string
+          example: researcher1@mailtrap.io
+        Role:
+          type: string
+          example: Researcher
+        Projects:
+          type: array
+          items:
+            type: string
+          example: [project1]
```
Comment (valyo) [inline dds_web/static/swagger.yaml:1061]:
```suggestion
        Unit:
          type: string
          example: ""
```

File: dds_web/static/swagger.yaml [hunk 3]
```diff
@@ -466,7 +466,339 @@ paths:
...
+                properties:
+                  invites:
+                    type: array
+                    items:
+                      $ref: "#/components/schemas/InvitesList"
+                  keys:
+                    type: array
+                    items:
+                      type: string
+                    example: [Email, Role, Projects, Created]
```
Comment (valyo) [inline dds_web/static/swagger.yaml:756]:
```suggestion
                    example: [Email, Unit, Role, Projects, Created]
```

File: dds_web/static/swagger.yaml [hunk 4]
```diff
@@ -466,7 +466,339 @@ paths:
...
+  /usage:
+    get:
+      tags:
+        - user
+        - project
```
Comment (valyo) [inline dds_web/static/swagger.yaml:761]: How is this endpoint called? I couldn't find anything in dds_cli calling it, neither in commands.py

**Review by valyo — CHANGES_REQUESTED** [review]

File: dds_web/static/swagger.yaml [hunk 5]
```diff
@@ -466,7 +466,339 @@ paths:
...
+                  example: researcher1@mailtrap.io
+  /user/activation:
+    post:
+      tags:
+        - user
+      summary: Unit Admins can reactivate/deactivate unitusers. Super admins can reactivate/deactivate any user.
+      operationId: userActivation
+      parameters:
+        - $ref: "#/components/parameters/defaultHeader"
+        - $ref: "#/components/parameters/projectID"
```
[the remark this answers is not in the document]
Comment (rv0lt) [reply dds_web/static/swagger.yaml:624]: true, missed to remove it

File: dds_web/static/swagger.yaml [hunk 6]
```diff
@@ -466,7 +466,339 @@ paths:
...
+              schema:
+                type: object
+                properties:
+                  email:
+                    type: string
+                    example: researcher1@mailtrap.io
+                  errors:
+                    type: array
+                    items:
+                      type: string
```
[the remark this answers is not in the document]
Comment (rv0lt) [reply dds_web/static/swagger.yaml:520]: Ok, for errors is true that it is return as a dic. For User Info, it seems to be returned as an array

image

File: dds_web/static/swagger.yaml (hunk 2 above)
Comment (rv0lt) [reply dds_web/static/swagger.yaml:1061]: Ok, so it seems that the returned json is different if using superadmin or unitadmin token.

When unitadmin no unit field is returned. 
Superadmin returns the empty unit filed (which is a bit weird)
I will see if I can mark it as optional returned or something like that

File: dds_web/static/swagger.yaml (hunk 3 above)
Comment (rv0lt) [reply dds_web/static/swagger.yaml:756]: See above

File: dds_web/static/swagger.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (valyo) [reply dds_web/static/swagger.yaml]: ah, of course, `emails_all` in user/info is a list, `array` should be a list in Swagger, right!

File: dds_web/static/swagger.yaml (hunk 2 above)
Comment (valyo) [reply dds_web/static/swagger.yaml:1063]: Yes, and it is a bit confusing, and as far as I can see it always returns an empty unit field for Superadmin, regardless 
of whether the invitation is connected to a unit or not. If it is connected, then only the project appears in the Projects field, but not the unit. This doesn't make a lot of sense, we either remove the unit field, or add a. value to it when relevant

Comment (valyo) [reply dds_web/static/swagger.yaml:1063]: I created a task for it

File: dds_web/static/swagger.yaml (hunk 4 above)
Comment (valyo) [reply dds_web/static/swagger.yaml:759]: Ok, I wasn't clear enough here: I meant what is making requests to this endpoint, which cli command or possibly backend command?

**Review by valyo — APPROVED** [review]

File: dds_web/static/swagger.yaml (hunk 4 above)
Comment (rv0lt) [reply dds_web/static/swagger.yaml:759]: Could happen that it is an endpoint created back in the day but forgotten. Could be deprecated in the future if we don't think it is good or usefull to have it

**Pull request merged**